packages/http/src/mocker/__tests__/HttpMocker.spec.ts

```diff
@@ -11,9 +11,9 @@ describe('HttpMocker', () => {
 
   beforeEach(() => {
     httpMocker = new HttpMocker(mockExampleGenerator);
-    jest.spyOn(helpers, 'negotiateOptionsForValidRequest').mockImplementation(jest.fn);
-    jest.spyOn(helpers, 'negotiateOptionsForInvalidRequest').mockImplementation(jest.fn);
-    jest.spyOn(mockExampleGenerator, 'generate').mockImplementation(jest.fn);
+    jest.spyOn(helpers, 'negotiateOptionsForValidRequest');
```

Comment:
Similar issue: AFAIK spyOn will not mock the implementation but use the actual implementation. What was the reason you changed this?

Reply:
Yeah, it won't mock the method by default, unless you instrument it do so by calling methods such as `mockImplementation` or `mockReturnValue` etc. etc.

Reply:
1. As mentioned above, the code is incorrect
2. Why mock something that just works?

Reply:
@XVincentX because it's a unit test. If there is one thing that I will strongly hold to is that unit test should **always mock/stub all** dependencies.

If `negotiateOptionsForValidRequest` stops working I don't want to see 100 tests files failing, but only the `helpers.spec.ts`. So unless it's an integration test (it's not in this case ;)) I don't see a reason to not mock that function.

Reply:
> If negotiateOptionsForValidRequest stops working I don't want to see 100 tests files failing, but only the helpers.spec.ts

I disagree but I have no forces and time right now to fight, so I'll make the change. To keep it short though tests should be designed to fail, not to pass. If a test passes every time and never fail, it tells me _nothing_. It should fail when *both* the dependencies *OR* the behaviour changes *OR* the dependency protocol changes.

Reply:
Restored in f8d018a

Reply:
I am currently very blocked on a hotfix because of the incredibly heavy amount of mocking involved in these very files. Unless the dependency is hitting the wire, the database or the filesystem (or something else similarly outside the control of the test suite), there is no reason to mock it. If you do not own it, definitely do not mock it. Mocking dependencies generally leads to code that does not work thinking it works really well. 

Please do not continue with the "mock all the things" approach or we will have a very rough time. If a test code is hitting so many dependencies that it is slow, it probably means our code is too tightly coupled, and shout be refactored.